src/Initializer/ParameterDB.h

```diff
@@ -116,16 +123,27 @@ class seissol::initializers::FaultGPGenerator : public seissol::initializers::Qu
   unsigned m_numberOfPoints;
 };
 
+template<class T>
 class seissol::initializers::ParameterDB {
-public:  
-  void addParameter(std::string const& parameter, double* memory, unsigned stride = 1) {
-    m_parameters[parameter] = std::make_pair(memory, stride);
-  }
-  void evaluateModel(std::string const& fileName, QueryGenerator const& queryGen);
+public: 
+  //for material parameters
+  void setMaterialType(seissol::model::MaterialType materialType) { m_materialType = materialType; }
+  void setMaterialVector(std::vector<T>* materials) { m_materials = materials; }
+  //for fault parameters
+  void addParameter(std::string const& parameter, double* memory, unsigned stride = 1) { m_parameters[parameter] = std::make_pair(memory, stride); }
   static bool faultParameterizedByTraction(std::string const& fileName);
+  //generic
+  void evaluateModel(std::string const& fileName, QueryGenerator const& queryGen);
   
 private:
+  //for material parameters
+  void addBindingPoints(easi::ArrayOfStructsAdapter<T> &adapter) {};
+  seissol::model::MaterialType m_materialType;
+  std::vector<T>* m_materials;
+  //for fault parameters
   std::unordered_map<std::string, std::pair<double*, unsigned>> m_parameters;
+  //generic
+  static easi::Component* loadModel(std::string const& fileName);
```

Comment:
If half of those are unused -> wouldn't it be better to split this into two different classes, one for material and one for faults? would maybe make the other functions clearer as well